syft/workers/base.py

```diff
@@ -163,6 +154,20 @@ def __init__(
                 # self is the to-be-created local worker
                 self.add_worker(self)
 
+        if hook is None:
+            self.framework = None
+        else:
+            # TODO[jvmancuso]: avoid branching here if possible, maybe by changing code in
+            #     execute_command or command_guard to not expect an attribute named "torch"
+            #     (#2530)
+            self.framework = hook.framework
```

Comment:
Had to move those here because of the following scenario.
1. In the tests we spin up the ```Websocket Server``` and ```Websocket Client```
2. Create one client and get a ```ConnectionRefused```
3. Create the second client and for this one ```line 149``` will run, which will override the ```remote``` attribute.
4. ```Remote``` contains the ```torch``` attribute (is a ```Torch``` instance - class that can be found in the hooks for the torch framework)  and the ```Torch``` instance contains an attribute ```worker``` that is a weak reference to ```self```.
The problem is that remote would get copied into the new ```WebsocketClient``` and we will have a reference to an item that should be garbage collected + ```ws``` attribute from ```WebsocketClient``` would be ```None``` (since we got a ```Connection Refused``` with the first client) --- there will be generated an error caused by the function ```_forward_message``` because ```self.remote.torch.worker.ws``` <--- the one that was refused is```None```